core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -185,6 +167,23 @@ public Integer getImagePullPauseTimeout() {
         return Integer.parseInt((String) properties.getOrDefault("pull.pause.timeout", "30"));
     }
 
+    /**
+     * Gets a configured setting from an environment variable (if present) or a configuration file property otherwise.
+     * @param propertyName name of configuration file property (dot-separated lower case)
+     * @return the found value, or null if not set
+     */
+    @Nullable
+    public String getEnvVarOrProperty(final String propertyName) {
```

Comment:
This is currently only used by the `PrefixingImageNameSubstitutor`, but I'm very tempted to use it for _most_ of the other properties that are configurable.

I think most of the properties would benefit from having an equivalent environment variable.

I know we discussed having the `reuse.enabled` as a config file property instead of an env var some time back, so perhaps we'd leave that as is?

Reply:
yes, let's keep `reuse.enabled` as an environment property 👍

Reply:
Would you be happy with every other property also being settable with an env var?

Reply:
I've done this as proposed.